### DISCUSSION TITLE
How to hide all custom console.logs inside code for production?

### DISCUSSION THREAD

**Discussion by JiaZhiheng** [nuxt/nuxt#16453] [Questions]
How to hide all custom console.logs inside code for production?
如何在生产环境中隐藏'console.log'？

**Comment by TheAlexLichter** [comment]
You can try using `vite.esbuild.drop`. See https://github.com/vitejs/vite/discussions/7920#discussioncomment-2709119

But be aware of possible side-effects.

**Reply by JiaZhiheng** [reply]
Thank you very much, but how to use 'vite' in 'Nuxt' project?
If I want to config vite, I should create "vite.config.js" file or write in "nuxt.config.ts" file ?

**Reply by TheAlexLichter** [reply]
You can use the top-level key `vite` in the `nuxt.config.ts`

**Reply by JiaZhiheng** [reply]
Your solved my problem again. Many thanks for your kind and warm help.

**Reply by TheAlexLichter** [reply]
No problem 🙌🏻

**Comment by arb85** [comment]
You can try to use a different logger, like [consola](https://github.com/unjs/consola) or [pino](https://getpino.io) and set different log levels in development and production (like -1 = silent)

**Reply by JiaZhiheng** [reply]
Thank you very much, it's a good idea!
